openml-h2o/src/main/java/com/feedzai/openml/h2o/algos/AbstractH2OAlgoUtils.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2019 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.feedzai.openml.h2o.algos;
+
+import com.feedzai.openml.data.schema.DatasetSchema;
+import hex.ModelBuilder;
+import org.apache.commons.lang3.StringUtils;
+import water.Job;
+import water.api.schemas3.ModelParametersSchemaV3;
+import water.fvec.Frame;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Abstract class to parse H2O algorithm params, validate them and train the model.
+ *
+ * @param <T> The concrete type of {@link ModelParametersSchemaV3 algorithm params}.
+ * @param <M> The concrete type of {@link ModelBuilder H20 Model Builder}.
+ * @since 0.1.0 //todo: validate if this should stay or be changed since this class went under serious changes (including name and generic param)
```

Comment:
If you only changed the location of file, IMO, you should keep the since as previous one, because the file already existed since 0.1.0. Then every extra method/function that you added should come with new since tag.
WDYT @paulojrp ?

Reply:
In this case it was because I also renamed it and changed the generic types.

Reply:
the original logic is basically the same

let's continue using the same version

also be aware that the version `@@@feedzai.next.release@@@` is a internal tag that will not work in this repo. You will need to specify the version manually